### PR TITLE
fix: [CI-4341]: fix invalid url for git connectors

### PR DESCRIPTION
**Pull request by smjt-h**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33106)

### CODE REVIEW

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33106)* status: 2 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java` line 403 at r1](https://reviewable.io/reviews/harness/harness-core/33106#-N2Bwgt10KMTzVChIU-8:-N2Bwgt10KMTzVChIU-9:bji3cp1) ([raw file](https://github.com/harness/harness-core/blob/46765002c179dc7c18c343649f8bfeeab22ff159/440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java#L403)):*
> ```Java
>     if (connectorType == ConnectorType.GIT || connectorType == ConnectorType.GITHUB
>         || connectorType == ConnectorType.GITLAB || connectorType == ConnectorType.BITBUCKET) {
>       ScmConnector scmConnector = (ScmConnector) connectorConfig;
> ```

This same logic would be required for the update flow too

**Comment by smjt-h** [comment]
> > Reviewed all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/harness/harness-core/33106)_ status: 2 file(s) to review. 1 change requests. 0 of 1 approvals obtained.
> 
> _[`440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java` line 401 at r1](https://reviewable.io/reviews/harness/harness-core/33106#-N2BwKRqEtub_mS8wfhG:-N2BwKRr3xsDXIhRQ1Mu:b-3t21h2) ([raw file](https://github.com/harness/harness-core/blob/46765002c179dc7c18c343649f8bfeeab22ff159/440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java#L401)):_
> 
> > ```java
> >     ConnectorConfigDTO connectorConfig = connectorInfo.getConnectorConfig();
> >     ConnectorType connectorType = connectorInfo.getConnectorType();
> >     if (connectorType == ConnectorType.GIT || connectorType == ConnectorType.GITHUB
> > ```
> 
> In every connector config, we can ovveride a validate() method. Can you please check that whether it would solve your issue. This way the solution would be more cleaner.
> 
> io/harness/delegate/beans/connector/vaultconnector/VaultConnectorDTO.java:119



> > Reviewed all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/harness/harness-core/33106)_ status: 2 file(s) to review. 1 change requests. 0 of 1 approvals obtained.
> 
> _[`440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java` line 401 at r1](https://reviewable.io/reviews/harness/harness-core/33106#-N2BwKRqEtub_mS8wfhG:-N2BwKRr3xsDXIhRQ1Mu:b-3t21h2) ([raw file](https://github.com/harness/harness-core/blob/46765002c179dc7c18c343649f8bfeeab22ff159/440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java#L401)):_
> 
> > ```java
> >     ConnectorConfigDTO connectorConfig = connectorInfo.getConnectorConfig();
> >     ConnectorType connectorType = connectorInfo.getConnectorType();
> >     if (connectorType == ConnectorType.GIT || connectorType == ConnectorType.GITHUB
> > ```
> 
> In every connector config, we can ovveride a validate() method. Can you please check that whether it would solve your issue. This way the solution would be more cleaner.
> 
> io/harness/delegate/beans/connector/vaultconnector/VaultConnectorDTO.java:119

Actually if I add it in every connectors override validate() method then exception occurs but the response is still success. Is there a specific file?

**Comment by smjt-h** [comment]
> > Reviewed all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/harness/harness-core/33106)_ status: 2 file(s) to review. 1 change requests. 0 of 1 approvals obtained.
> 
> _[`440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java` line 401 at r1](https://reviewable.io/reviews/harness/harness-core/33106#-N2BwKRqEtub_mS8wfhG:-N2BwKRr3xsDXIhRQ1Mu:b-3t21h2) ([raw file](https://github.com/harness/harness-core/blob/46765002c179dc7c18c343649f8bfeeab22ff159/440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java#L401)):_
> 
> > ```java
> >     ConnectorConfigDTO connectorConfig = connectorInfo.getConnectorConfig();
> >     ConnectorType connectorType = connectorInfo.getConnectorType();
> >     if (connectorType == ConnectorType.GIT || connectorType == ConnectorType.GITHUB
> > ```
> 
> In every connector config, we can ovveride a validate() method. Can you please check that whether it would solve your issue. This way the solution would be more cleaner.
> 
> io/harness/delegate/beans/connector/vaultconnector/VaultConnectorDTO.java:119



> > Reviewed all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/harness/harness-core/33106)_ status: 2 file(s) to review. 1 change requests. 0 of 1 approvals obtained.
> 
> _[`440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java` line 401 at r1](https://reviewable.io/reviews/harness/harness-core/33106#-N2BwKRqEtub_mS8wfhG:-N2BwKRr3xsDXIhRQ1Mu:b-3t21h2) ([raw file](https://github.com/harness/harness-core/blob/46765002c179dc7c18c343649f8bfeeab22ff159/440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java#L401)):_
> 
> > ```java
> >     ConnectorConfigDTO connectorConfig = connectorInfo.getConnectorConfig();
> >     ConnectorType connectorType = connectorInfo.getConnectorType();
> >     if (connectorType == ConnectorType.GIT || connectorType == ConnectorType.GITHUB
> > ```
> 
> In every connector config, we can ovveride a validate() method. Can you please check that whether it would solve your issue. This way the solution would be more cleaner.
> 
> io/harness/delegate/beans/connector/vaultconnector/VaultConnectorDTO.java:119

Actually if I add it in every connectors override validate() method then exception occurs but the response is still success. Is there a specific file?

**Comment by smjt-h** [comment]
> > _[Reviewable](https://reviewable.io/reviews/harness/harness-core/33106)_ status: 2 file(s) to review. 1 change requests. 0 of 1 approvals obtained.
> 
> _[`440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java` line 403 at r1](https://reviewable.io/reviews/harness/harness-core/33106#-N2Bwgt10KMTzVChIU-8:-N2Bwgt10KMTzVChIU-9:bji3cp1) ([raw file](https://github.com/harness/harness-core/blob/46765002c179dc7c18c343649f8bfeeab22ff159/440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java#L403)):_
> 
> > ```java
> >     if (connectorType == ConnectorType.GIT || connectorType == ConnectorType.GITHUB
> >         || connectorType == ConnectorType.GITLAB || connectorType == ConnectorType.BITBUCKET) {
> >       ScmConnector scmConnector = (ScmConnector) connectorConfig;
> > ```
> 
> This same logic would be required for the update flow too



> > _[Reviewable](https://reviewable.io/reviews/harness/harness-core/33106)_ status: 2 file(s) to review. 1 change requests. 0 of 1 approvals obtained.
> 
> _[`440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java` line 403 at r1](https://reviewable.io/reviews/harness/harness-core/33106#-N2Bwgt10KMTzVChIU-8:-N2Bwgt10KMTzVChIU-9:bji3cp1) ([raw file](https://github.com/harness/harness-core/blob/46765002c179dc7c18c343649f8bfeeab22ff159/440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java#L403)):_
> 
> > ```java
> >     if (connectorType == ConnectorType.GIT || connectorType == ConnectorType.GITHUB
> >         || connectorType == ConnectorType.GITLAB || connectorType == ConnectorType.BITBUCKET) {
> >       ScmConnector scmConnector = (ScmConnector) connectorConfig;
> > ```
> 
> This same logic would be required for the update flow too

I have added logic for update flow as well and tested

**Comment by smjt-h** [comment]
> > _[Reviewable](https://reviewable.io/reviews/harness/harness-core/33106)_ status: 2 file(s) to review. 1 change requests. 0 of 1 approvals obtained.
> 
> _[`440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java` line 403 at r1](https://reviewable.io/reviews/harness/harness-core/33106#-N2Bwgt10KMTzVChIU-8:-N2Bwgt10KMTzVChIU-9:bji3cp1) ([raw file](https://github.com/harness/harness-core/blob/46765002c179dc7c18c343649f8bfeeab22ff159/440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java#L403)):_
> 
> > ```java
> >     if (connectorType == ConnectorType.GIT || connectorType == ConnectorType.GITHUB
> >         || connectorType == ConnectorType.GITLAB || connectorType == ConnectorType.BITBUCKET) {
> >       ScmConnector scmConnector = (ScmConnector) connectorConfig;
> > ```
> 
> This same logic would be required for the update flow too



> > _[Reviewable](https://reviewable.io/reviews/harness/harness-core/33106)_ status: 2 file(s) to review. 1 change requests. 0 of 1 approvals obtained.
> 
> _[`440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java` line 403 at r1](https://reviewable.io/reviews/harness/harness-core/33106#-N2Bwgt10KMTzVChIU-8:-N2Bwgt10KMTzVChIU-9:bji3cp1) ([raw file](https://github.com/harness/harness-core/blob/46765002c179dc7c18c343649f8bfeeab22ff159/440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java#L403)):_
> 
> > ```java
> >     if (connectorType == ConnectorType.GIT || connectorType == ConnectorType.GITHUB
> >         || connectorType == ConnectorType.GITLAB || connectorType == ConnectorType.BITBUCKET) {
> >       ScmConnector scmConnector = (ScmConnector) connectorConfig;
> > ```
> 
> This same logic would be required for the update flow too

I have added logic for update flow as well and tested

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33106)* status: 2 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java` line 401 at r1](https://reviewable.io/reviews/harness/harness-core/33106#-N2BwKRqEtub_mS8wfhG:-N2GPWIg6Y4xv3OHN90c:b-iu6zh6) ([raw file](https://github.com/harness/harness-core/blob/46765002c179dc7c18c343649f8bfeeab22ff159/440-connector-nextgen/src/main/java/io/harness/connector/impl/DefaultConnectorServiceImpl.java#L401)):*
<details><summary><i>Previously, smjt-h wrote…</i></summary><blockquote>

Actually if I add it in every connectors override validate() method then exception occurs but the response is still success. Is there a specific file?
</blockquote></details>

Can you share the response ?

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33106)* status: 8 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/awscodecommit/AwsCodeCommitConnectorDTO.java` line 48 at r3](https://reviewable.io/reviews/harness/harness-core/33106#-N2Kg7R94CpYzQG-DF05:-N2Kg7R94CpYzQG-DF06:bytgiyh) ([raw file](https://github.com/harness/harness-core/blob/5d6981f026420b1f6604277902f2035099d5f2f4/954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/awscodecommit/AwsCodeCommitConnectorDTO.java#L48)):*
> ```Java
>   @Valid @NotNull AwsCodeCommitAuthenticationDTO authentication;
>   Set<String> delegateSelectors;
>   ScmHelper scmHelper;
> ```

Since we are not using scmHelper, we can remove it from here.

We  are directly calling the utility function

**Review by DeepakPatankar — COMMENTED** [review]
> Reviewed 3 of 8 files at r3, 5 of 5 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33106)* status: 1 change requests. 0 of 1 approvals obtained.

**Review by DeepakPatankar — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33106)* status: :shipit: complete! ✓

**Review by harshjain12 — CHANGES_REQUESTED** [review]
UTs are missing

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33106#-:-N2Kvm_CECDSIvcaQHcJ:bp8vs09)* status: :shipit: complete! ✓

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/helper/ScmHelper.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+package io.harness.delegate.beans.connector.scm.helper;
+
+import io.harness.exception.InvalidRequestException;
+
+import lombok.extern.slf4j.Slf4j;
+import org.jetbrains.annotations.NotNull;
+
+@Slf4j
+public class ScmHelper {
```
Comment (harshjain12) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/helper/ScmHelper.java:16]: do not create new helper
add this in git client helper.

Comment (smjt-h) [reply 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/scm/helper/ScmHelper.java]: done

**Pull request merged**